### PR TITLE
Fixes #26782 - Update Rubocop to 0.71.0

### PR DESCRIPTION
**Pull request by mmoll**

### CODE REVIEW

**Comment by mmoll** [comment]
rebased

**Comment by mmoll** [comment]
rebased and updated.

File: bundler.d/development.rb [hunk 1]
```diff
@@ -1,6 +1,8 @@
 group :development do
   gem 'maruku', '~> 0.7'
-  gem 'rubocop', '~> 0.64.0'
+  gem 'rubocop', '~> 0.71.0'
```
Comment (mmoll) [inline bundler.d/development.rb]: BTW, maybe this should go into `bundler.d/test.rb`?

**Comment by mmoll** [comment]
rebased.

File: bundler.d/development.rb (hunk 1 above)
Comment (ezr-ondrej) [reply bundler.d/development.rb]: Maybe someone had in mind that it helps with the development...
But I agree to move... isn't it here for some local requires to work? If it works after move, let's do it :)

**Review by ezr-ondrej — DISMISSED** [review]
Tests are passing, Hound is ok, jenkins is ok, locally it works.
Some issues with merging?

File: bundler.d/development.rb (hunk 1 above)
Comment (mmoll) [reply bundler.d/development.rb]: Then let me change this in this PR, minimizes the diff...

**Comment by ezr-ondrej** [comment]
[test katello]

**Comment by mmoll** [comment]
Should be all fine now :bowtie:

File: .rubocop_todo.yml [hunk 2]
```diff
@@ -176,7 +184,18 @@ Naming/MethodName:
...
   Enabled: false
 
-# Offense count: 144
+# Offense count: 60
+# Cop supports --auto-correct.
+# Configuration parameters: PreferredName.
+Naming/RescuedExceptionsVariableName:
+  Enabled: false
+
+# Offense count: 1
```
Comment (ekohl) [inline .rubocop_todo.yml:193]: Would it be easier to fix this particular case or disable the cop inline?

Comment (mmoll) [reply .rubocop_todo.yml:193]: I'd like to fix this after the merge.

**Review by ekohl — APPROVED** [review]

**Review by tbrisker — APPROVED** [review]
Thanks @mmoll !

File: .rubocop_todo.yml (hunk 2 above)
Comment (mmoll) [reply .rubocop_todo.yml:193]: GH-6887

**Pull request merged**